# Breast Cancer Prediction
Predict whether a tumor is **malignant** or **benign** using Logistic Regression.

- Dataset: Wisconsin Breast Cancer (569 patients, 30 features)
- Model: Logistic Regression

## Step 1: Import libraries

In [1]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix

## Step 2: Load data
`0` = malignant (cancer), `1` = benign (no cancer)

In [2]:
data = load_breast_cancer()
X = data.data
y = data.target
print('Total patients:', X.shape[0])
print('Total features:', X.shape[1])

Total patients: 569
Total features: 30


## Step 3: Split into train (80%) and test (20%)

In [3]:
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42)

## Step 4: Scale the features
So that all features are on the same range.

In [4]:
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

## Step 5: Train the model

In [5]:
model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)

LogisticRegression(max_iter=1000)

## Step 6: Predict on test data

In [6]:
y_pred = model.predict(X_test)

## Step 7: Check results

In [7]:
print('Accuracy:', round(accuracy_score(y_test, y_pred) * 100, 2), '%')
print('Confusion Matrix:')
print(confusion_matrix(y_test, y_pred))

Accuracy: 97.37 %
Confusion Matrix:
[[41  2]
 [ 1 70]]


## Conclusion
The Logistic Regression model reached about **97% accuracy** on unseen data. Only 3 out of 114 test patients were misclassified.